lib/util.js

```diff
@@ -132,7 +144,144 @@ const tryStatFile = filePath => {
 
 const fixArgs = args => (args ? (typeof args === 'string' ? args.split(' ') : args) : []);
 
+const isPlainObject = value => {
+  if (typeof value !== 'object' || value === null) return false;
+
+  if (Object.prototype.toString.call(value) !== '[object Object]') return false;
+
+  const proto = Object.getPrototypeOf(value);
+  if (proto === null) return true;
+
+  const Ctor = Object.prototype.hasOwnProperty.call(proto, 'constructor') && proto.constructor;
+  return (
+    typeof Ctor === 'function' &&
+    Ctor instanceof Ctor &&
+    Function.prototype.call(Ctor) === Function.prototype.call(value)
+  );
+};
+
+const get = (obj, propsArg, defaultValue) => {
+  if (!obj) {
+    return defaultValue;
+  }
+  var props, prop;
+  if (Array.isArray(propsArg)) {
+    props = propsArg.slice(0);
+  }
+  if (typeof propsArg == 'string') {
+    props = propsArg.split('.');
+  }
+  if (typeof propsArg == 'symbol') {
+    props = [propsArg];
+  }
+  if (!Array.isArray(props)) {
+    if (typeof defaultValue !== 'undefined') return defaultValue;
+    throw new Error('props arg must be an array, a string or a symbol');
+  }
+  while (props.length) {
+    prop = props.shift();
+    if (!obj) {
+      return defaultValue;
+    }
+    obj = obj[prop];
+    if (obj === undefined) {
+      return defaultValue;
+    }
+  }
+  return obj;
+};
+
+const isObject = value => {
+  var type = typeof value;
+  return value != null && (type == 'object' || type == 'function');
+};
+
+const upperFirst = string => {
+  return string ? string.charAt(0).toUpperCase() + string.slice(1) : '';
+};
+
+const castArray = arr => {
+  return Array.isArray(arr) ? arr : [arr];
+};
+
+const before = (n, func) => {
+  var result;
+  if (typeof func != 'function') {
+    throw new TypeError('Missing argument for `func`');
+  }
+  n = parseInt(n);
+  return function () {
+    if (--n > 0) {
+      result = func.apply(this, arguments);
+    }
+    if (n <= 1) {
+      func = undefined;
+    }
+    return result;
+  };
+};
+
+const once = fn => {
+  return before(2, fn);
+};
+
+const pick = (object, keys) => {
+  return keys.reduce((obj, key) => {
+    if (object && Object.prototype.hasOwnProperty.call(object, key)) {
+      obj[key] = object[key];
+    }
+    return obj;
+  }, {});
+};
+
+const merge = (target, ...sources) => {
```

Comment:
Removing a fairly large dependency like Lodash is great, but from a maintainer's perspective inlining such non-trivial functions isn't great. What do you think about using small focused dependencies? Or something else that does not require me to maintain this myself?

Reply:
in my experience and view, if a dependency is too small, do not use it as a dep/create a dep for it. Please see the pr comment: https://github.com/release-it/release-it/pull/1202#issuecomment-2684236494

But, we are different people with different views, if you think that splitting up in minor/small dependencies is the way to go, that is your call :)